auction-house/program/src/receipt/mod.rs

```diff
@@ -328,3 +327,29 @@ pub fn close_public_bid_receipt<'info>(
 
     Ok(())
 }
+
+
+#[derive(Accounts)]
+#[instruction(receipt_bump: u8)]
+pub struct PrintPurchaseReceipt<'info> {
+    #[account(init, seeds=[PURCHASE_RECEIPT_PREFIX.as_bytes(), seller_trade_state.key().as_ref(), buyer_trade_state.key().as_ref()], bump=receipt_bump, payer=bookkeeper, space=PURCHASE_RECEIPT_SIZE)]
+    receipt: Account<'info, PurchaseReceipt>,
+    seller_trade_state: UncheckedAccount<'info>,
+    buyer_trade_state: UncheckedAccount<'info>,
+    #[account(mut)]
+    bookkeeper: Signer<'info>,
+    system_program: Program<'info, System>,
+    rent: Sysvar<'info, Rent>,
+    clock: Sysvar<'info, Clock>,
+    instruction: UncheckedAccount<'info>,
+}
+
+pub fn print_purchase_receipt<'info>(
+  ctx: Context<'_, '_, '_, 'info, PrintPurchaseReceipt<'info>>,
+  receipt_bump: u8,
+) -> ProgramResult {
+
+let prev_instruction_data = solana_program::sysvar::instructions::get_instruction_relative(-1, &ctx.accounts.instruction)?;
```

Comment:
1. What is the best way to access the instructions program account info? Right now im passing `sysvar::instructions::id()` as unchecked account.

2. I have access to the previous instruction data. How do I deserialize it into the accounts::ExecuteSale and `instruction::ExecuteSale`?

Reply:
_I may be off what you were requesting. Do you want me to assert the execute sale instruction is part of the instruction or read the contents of the instruction for relevant accounts? The latter is what I'm striving for but still not seeing how to take the instruction into the anchor accounts struct.

I'm looking into this but can't find any examples of it used other than within anchor lang codegen.

https://docs.rs/anchor-lang/0.16.1/anchor_lang/trait.Accounts.html#tymethod.try_accounts